core/src/main/java/org/infinispan/transaction/impl/WriteSkewHelper.java

```diff
@@ -46,7 +47,12 @@ public static EntryVersionsMap performWriteSkewCheckAndReturnNewVersions(Version
       for (WriteCommand c : prepareCommand.getModifications()) {
          for (Object k : c.getAffectedKeys()) {
             if (ksl.performCheckOnKey(k)) {
-               ClusteredRepeatableReadEntry entry = (ClusteredRepeatableReadEntry) context.lookupEntry(k);
+               CacheEntry cacheEntry = context.lookupEntry(k);
+               if (!(cacheEntry instanceof VersionedRepeatableReadEntry)) {
+                  // DeltaAware?
```

Comment:
DeltaAware and optimistic+write-skew isn't implemented. you don't have to worry about it